rewrite-java/src/main/java/org/openrewrite/java/internal/template/AnnotationTemplateGenerator.java

```diff
@@ -0,0 +1,237 @@
+                    after.append("return ")
+                            .append(valueOfType(m.getReturnTypeExpression().getType()))
+                            .append(";\n");
+                }
+
+                before.insert(0, m.withBody(null)
+                        .withLeadingAnnotations(emptyList())
+                        .withPrefix(Space.EMPTY)
+                        .printTrimmed().trim() + '{');
+                after.append('}');
+            } else if (parent instanceof J.Block) {
+                J.Block b = (J.Block) j;
+
+                // variable declarations up to the point of insertion
+                for (Statement statement : b.getStatements()) {
+                    if (statement == prior) {
+                        break;
+                    } else if (statement instanceof J.VariableDeclarations) {
+                        J.VariableDeclarations v = (J.VariableDeclarations) statement;
+                        if (v.hasModifier(J.Modifier.Type.Final)) {
+                            before.insert(0, "\n" + variable(v) + ";\n");
+                        }
+                    }
+                }
+
+                before.insert(0, "{\n");
+                if (b.isStatic()) {
+                    before.insert(0, "static");
+                }
+                after.append('}');
+            }
+        } else if (j instanceof J.VariableDeclarations) {
+            J.VariableDeclarations v = (J.VariableDeclarations) j;
+            if (v.hasModifier(J.Modifier.Type.Final)) {
+                before.insert(0, variable((J.VariableDeclarations) j) + '=');
+            }
+        } else if (j instanceof J.NewClass) {
+            J.NewClass n = (J.NewClass) j;
+            n = n.withBody(null).withPrefix(Space.EMPTY);
+            before.insert(0, '{');
+            before.insert(0, n.printTrimmed().trim());
+            after.append("};");
+        }
+
+        template(next(cursor), j, before, after);
+    }
+
+    private void classDeclaration(StringBuilder before, J.ClassDeclaration parent) {
+        J.ClassDeclaration c = parent;
+        for (Statement statement : c.getBody().getStatements()) {
+            if (statement instanceof J.VariableDeclarations) {
+                J.VariableDeclarations v = (J.VariableDeclarations) statement;
+                if (v.hasModifier(J.Modifier.Type.Final) && v.hasModifier(J.Modifier.Type.Static)) {
+                    before.insert(0, variable((J.VariableDeclarations) statement) + ";\n");
+                }
+            } else if (statement instanceof J.ClassDeclaration) {
+                // this is a sibling class. we need declarations for all variables and methods.
+                // setting prior to null will cause them all to be written.
+                before.insert(0, '}');
+                classDeclaration(before, (J.ClassDeclaration) statement);
+            }
+        }
+        c = c.withBody(null).withLeadingAnnotations(null).withPrefix(Space.EMPTY);
```

Comment:
This might be causing NPE in KotlinPrinter: https://github.com/openrewrite/rewrite-static-analysis/issues/234
Should [Kotlin printer](https://github.com/openrewrite/rewrite-kotlin/blob/666bc77a4be8b4252332117b6a4103085902df7c/src/main/java/org/openrewrite/kotlin/internal/KotlinPrinter.java#L722) add a guard for `null` body?

Reply:
Calling `withBody()` with `null` is wrong. I've changed this in https://github.com/openrewrite/rewrite/commit/c625f30ca32f8af7a196dd3b00b720919db22dcf.